src/main/java/com/ibm/eventstreams/connect/mqsource/builders/ForceSchemaJsonBuilder.java

```diff
@@ -0,0 +1,116 @@
+    private static final Logger log = LoggerFactory.getLogger(ForceSchemaJsonBuilder.class);
+
+    private JsonConverter converter;
+
+    private ObjectNode envelope;
+    private static ObjectMapper MAPPER = new ObjectMapper();
+    private String registryURL;
+
+    public ForceSchemaJsonBuilder() {
+        log.info("Building records using com.ibm.eventstreams.connect.mqsource.builders.ForceSchemaJsonBuilder");
+        converter = new JsonConverter();
+        HashMap<String, String> m = new HashMap<>();
+        m.put("schemas.enable", "true");
+        converter.configure(m, false);
+    }
+
+    @Override
+    public void configure(Map<String, String> props) {
+        super.configure(props);
+        registryURL = props.get("value.converter.schema.registry.url");
+    }
+
+    /**
+     * Extracts JSON value from JMS, wraps with schema according to topic and deserializes into connect data struct
+     *
+     * @param context        the JMS context to use for building messages
+     * @param topic          the Kafka topic
+     * @param messageBodyJms whether to interpret MQ messages as JMS messages
+     * @param message        the message
+     * @return the Kafka Connect SourceRecord's value
+     * @throws JMSException Message could not be converted
+     */
+    @Override
+    SchemaAndValue getValue(JMSContext context, String topic, boolean messageBodyJms, Message message) throws JMSException {
+        if (envelope == null) {
+            envelope = resolveSchemaEnvelope(topic);
+            log.debug("JSON Schema resolved for topic {}", topic);
+        }
+        String payload = extractPayload(message);
+        envelope.putRawValue("payload", new RawValue(payload));
+        return tryConvertToJSON(topic);
+    }
+
+    private String extractPayload(Message message) throws JMSException {
+        if (message instanceof BytesMessage) {
+            return new String(message.getBody(byte[].class), StandardCharsets.UTF_8);
+        } else if (message instanceof TextMessage) {
+            return message.getBody(String.class);
+        } else {
+            log.error("Unsupported JMS message type {}", message.getClass());
+            throw new ConnectException("Unsupported JMS message type");
+        }
+    }
+
+    private SchemaAndValue tryConvertToJSON(String topic) throws MessageFormatException {
+        try {
+            byte[] jsonBytes = MAPPER.writeValueAsBytes(envelope);
+            return converter.toConnectData(topic, jsonBytes);
+        } catch (JsonProcessingException exception) {
+            throw new MessageFormatException("Message is not serializable", exception.toString());
+        }
+    }
+
+    /**
+     * Creates simple JSON envelope with schema, but without payload
+     */
+    private ObjectNode resolveSchemaEnvelope(String topic) throws JMSException {
+        try {
+            CachedSchemaRegistryClient cachedSchemaRegistryClient = new CachedSchemaRegistryClient(registryURL, 1);
```

Comment:
I think I would provide some meaningful constants for those magic numbers

- (registryURL, 1) doesn't really say much
- new AvroData(1) is the same

Reply:
yeah I am agreed with Michal about that 1, but for all is ok for me